javascript/extractor/src/com/semmle/js/extractor/ScriptExtractor.java

```diff
@@ -78,4 +96,40 @@ public LoCInfo extract(TextualExtractor textualExtractor) {
 
     return loc;
   }
+
+  /**
+   * A minimal model of `package.json` files that can be used to read the "type" field.
+   */
+  private static class PackageJSON {
+    String type;
+  }
+
+  /**
+   * Returns the "type" field from the nearest `package.json` file (searching up the file hierarchy).
+   */
+  private String getPackageType(File folder) {
+    if (folder == null || !folder.isDirectory()) {
+      return null;
+    }
+    if (packageTypeCache.containsKey(folder)) {
+      return packageTypeCache.get(folder).orElse(null);
+    }
+    File file = new File(folder, "package.json");
+    if (file.isDirectory()) {
+      return null;
+    }
+    if (!file.exists()) {
+      String result = getPackageType(folder.getParentFile());
+      packageTypeCache.put(folder, Optional.ofNullable(result));
+      return result;
+    }
+    try {
+      BufferedReader reader = new BufferedReader(new FileReader(file));
+      String result = new Gson().fromJson(reader, PackageJSON.class).type;
```

Comment:
Could you add a test where `type` is present but is not a string or `null` (e.g a number). It seems like this will crash the extractor.

Also catch `JsonSyntaxException` below in case the file has a syntax error.

Reply:
`JsonSyntaxException` seems to also catch the case where `type` is not a string.